server/worker/worker.go

```diff
@@ -0,0 +1,244 @@
+package worker
+
+import (
+	"encoding/json"
+	"github.com/go-kit/kit/log/level"
+)
+
+type Dashboard struct {
+	Uid   string
+	Data  map[string]interface{}
+	Title string
+}
+
+type DashboardsResponse struct {
+	Uid string `json:"uid"`
+}
+
+type DashboardResponse struct {
+	Dashboard map[string]interface{} `json:"dashboard"`
+}
+
+type DashboardUpdateResponse struct {
+	FolderUid string `json:"folderUid"`
+	Id        uint64 `json:"id"`
+	Slug      string `json:"slug"`
+	Status    string `json:"status"`
+	Uid       string `json:"uid"`
+	Url       string `json:"url"`
+	Version   uint64 `json:"version"`
+}
+
+func getTypedDashboardData(rawDashboardData *Dashboard) map[string]interface{} {
+	dashboard, ok := rawDashboardData.Data["dashboard"]
+	var emptyDashboardData map[string]interface{}
+	if !ok {
+		return emptyDashboardData
+	}
+
+	dashboardData, ok := dashboard.(map[string]interface{})
+	if !ok {
+		return emptyDashboardData
+	}
+
+	return dashboardData
+}
+
+func getTypedPanelsData(dashboardData map[string]interface{}) []interface{} {
+	var emptyResponse []interface{}
+
+	// Dashboards data might contain folders or other nonusable "dashboards"
+	// All usable dashboards have minimum of empty panels array, even when dashboard is newly created without any panels
+	// Due to this, we do not handle non "ok"
+	panels, ok := dashboardData["panels"]
+	if !ok {
+		return emptyResponse
+	}
+
+	panelList, ok := panels.([]interface{})
+	if !ok {
+		return emptyResponse
+	}
+
+	return panelList
+}
+
+func isAnalyticsPanel(panel map[string]interface{}) bool {
+	panelType, ok := panel["type"]
+	if ok {
+		return panelType == "macropower-analytics-panel"
+	}
+
+	return false
+}
+
+func (api *Client) GetDashboard(uid string) *Dashboard {
+	res, err := api.Get("/api/dashboards/uid/" + uid)
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "GetDashboard - api.Get failed",
+			"error", err,
+		)
+
+		return nil
+	}
+
+	var dashboardData map[string]interface{}
+	err = json.Unmarshal(res, &dashboardData)
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "GetDashboard - Failed to parse dashboardData response",
+			"error", err,
+		)
+
+		return nil
+	}
+
+	return &Dashboard{
+		Uid:  uid,
+		Data: dashboardData,
+	}
+}
+
+func (api *Client) updateDashboards(dashboards []Dashboard) {
+	for _, dashboard := range dashboards {
+		if dashboard.Title == "test dashboard - empty" {
```

Comment:
I guess this needs to be removed, otherwise it will only update test dashboard. You could also extract this into a dashboard filtering functionality which could be set via cli arg as comma separated list for example.